cli/src/commands/rebase.rs

```diff
@@ -433,6 +485,85 @@ fn rebase_revision(
     }
 }
 
+fn rebase_revision_after(
```

Comment:
There is quite a lot of new code below. I wonder if I should try to finish my rebase API rewrite before we merge this PR and then this can hopefully be simplified. I haven't even thought about how that would, however.

I'll spend some more time on that refactoring now. I think I'm getting pretty close.

Reply:
Yeah, I'm happy to defer reviewing/merging this until after the refactor is done, since it might be able to simplify this PR as well.

Reply:
That's now done so you can try to use the new API for this feature. It might be best to first rewrite the existing `rebase -r` code to use the new API.